core/vm/environment.go

```diff
@@ -35,6 +37,7 @@ type (
 
 // Context provides the EVM with auxilary information. Once provided it shouldn't be modified.
 type Context struct {
+	context.Context
```

Comment:
One thing I don't like about the current EVM is that it may only be used once, this is completely opaque and only becomes apparent when one inspects the source code or reads the documentation. This becomes apparent whit something like a `Cancel` method is added which sets an atomic field as it it will also introduce weird race conditions.

The main problem is that when the EVM is cancelled and the `abort` flag is set you'll no longer we able to run any other program, which imo kind of sucks.